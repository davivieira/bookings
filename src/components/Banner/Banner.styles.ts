import { Box, styled, Typography } from "@mui/material";

export const SCBannerContainer = styled(Box)(({ theme: { breakpoints, palette } }) => ({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  textAlign: 'center',
  justifyContent: 'center',
  minHeight: 400,
  width: '100%',
  background: `linear-gradient(to right, ${palette.primary.main}, ${palette.secondary.main})`,
  [breakpoints.down('lg')]: {
    left: 0,
    right: 0,
    width: '100vw'
  }
}))

export const SCSlogan = styled(Typography)(() => ({
  color: 'white',
  fontSize: 60,
  fontWeight: 700
}))