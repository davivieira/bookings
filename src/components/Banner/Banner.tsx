import { SCBannerContainer, SCSlogan } from "./Banner.styles"

function Banner() {
  return (
    <SCBannerContainer>
      <SCSlogan>Rent your next stay</SCSlogan>
    </SCBannerContainer>
  )
}

export default Banner