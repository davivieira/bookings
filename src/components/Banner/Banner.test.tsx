import { render, screen } from "@testing-library/react";
import Banner from "./Banner";

describe("Banner", () => {
  it("renders slogan correctly", () => {
    render(<Banner />);
    const sloganElement = screen.getByText(/Rent your next stay/i);
    expect(sloganElement).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<Banner />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
