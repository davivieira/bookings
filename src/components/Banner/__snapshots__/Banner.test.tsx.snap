// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner matches snapshot 1`] = `
<div
  class="MuiBox-root css-1q8qt3h"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1snziej-MuiTypography-root"
  >
    Rent your next stay
  </p>
</div>
`;
